src/sysext/sysext.c

```diff
@@ -516,6 +594,95 @@ static int mount_overlayfs(
         return 0;
 }
 
+static char *hierarchy_as_single_path_component(const char *hierarchy) {
+        /* We normally expect hierarchy to be /usr, /opt or /etc, but for debugging purposes the hierarchy
+         * could very well be like /foo/bar/baz/. So for a given hierarchy we generate a directory name by
+         * stripping the leading and trailing separators and replacing the rest of separators with dots. This
+         * makes the generated name to be the same for /foo/bar/baz and for /foo/bar.baz, but, again,
+         * speciyfing a different hierarchy is a debugging feature, so non-unique mapping should not be an
+         * issue in general case. */
+        const char *stripped = hierarchy;
+        _cleanup_free_ char *dir_name = NULL;
+
+        assert(hierarchy);
+
+        while (*stripped == '/')
+                stripped++;
+
+        dir_name = strdup(stripped);
+        if (!dir_name)
+                return NULL;
+        dir_name = delete_trailing_chars(dir_name, "/");
```

Comment:
this just returns the first arg, hence the assignment here is pretty much redundant. I'd drop it. i.e. just call delete_trailing_chars() as if it had a void return value